### PR TITLE
Filter books

### PR DESCRIPTION
**Pull request by MiguelDP4**
- Add filter reducer
- Add CHANGE_FILTER action
- Add constant file, create constants for all strings and arrays
- Create CategoryFilter component
- Map dispatch to props on CategoryFilter
- Add form validation, books need title and category

### CODE REVIEW

File: src/reducers/filters.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { CHANGE_FILTER } from '../constants';
+
+const initialFilterState = '';
```
Comment (whiz25) [inline src/reducers/filters.js]: - this should default to 'ALL' category

File: src/components/CategoryFilter.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { CATEGORIES } from '../constants';
+
+export default function CategoryFilter(props) {
+  const { handleChange } = props;
+  return (
+    <select onChange={handleChange}>
+      {CATEGORIES.map(category => (
```
Comment (whiz25) [inline src/components/CategoryFilter.js]: - merge 'All' option with available categories. 
```javascript
const categories = [...['All'], ...CATEGORIES]
```

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required :pray: 
You have done well in this PR. Check my comments for suggestions on how to improve the PR.
- please make sure to also fix the linter errors.

### Martin Afani

File: src/components/CategoryFilter.js (hunk 2 above)
Comment (fedgut) [reply src/components/CategoryFilter.js:9]: image

File: src/reducers/filters.js (hunk 1 above)
Comment (fedgut) [reply src/reducers/filters.js]: image

Comment (fedgut) [reply src/reducers/filters.js]: Hello! Thank you for your observations. The pathwright actually tells us to keep the ALL category **outside**  the react store!

**Comment by fedgut** [comment]
**Hello @NEXT TSE!** 

We want to make sure the code review is not conflicting with the instructions in pathwright! 

image

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

- [ ] To answer your question, the code review does not conflict with the instructions.

> You can implement this as advised by the first reviewer. You are saving `All books` as a part of your `CATEGORIES` and this should not be because `All books` is not a category. You should merge the book categories with `All books` instead as mentioned in the instruction. Let's fix this.

**KIndly submit another CRR once these issues have been solved.** _Happy coding!_ 🎉

**Review by Forison — APPROVED** [review]
## Status:: Approved✔️.
# 🙋‍♂️ Team
### Good job on implementing the changes raised in the previous reviews🤗.
### Go ahead and proceed to the next task🤗.
#### Happy coding👨‍💻
Addo Forison

**Pull request merged**